### ISSUE TITLE
t1*-read-tree official tests

### ISSUE THREAD

**Issue by sirnewton01** [driusan/dgit#112]
Make the majority of these tests from the official suite pass in order to ensure that merging is working properly.

**Comment by driusan** [comment]
Missing are:

t1004-read-tree-m-u-wf.sh (implemented, but partially ignored because it uses some missing subcommands)
t1005-read-tree-reset.sh
t1008-read-tree-overlay.sh
t1011-read-tree-sparse-checkout.sh
t1012-read-tree-df.sh
t1013-read-tree-submodule.sh
t1014-read-tree-confusing.sh

**Comment by driusan** [comment]
#102 is required for t1004-read-tree-m-u-wf.sh (at least)

**Comment by driusan** [comment]
t1011-read-tree-sparse-checkout.sh depends on #121. t1008-read-tree-overlay.sh depends on figuring out the answer to https://stackoverflow.com/questions/52175720/how-does-git-read-tree-work-without-m-or-reset-option

**Comment by driusan** [comment]
I think we can call this resolved. The submodule test deserves its own issue,  and other than that there's only three minor subtests skipped because unrelated commands/options are missing.
